crates/re_arrow_store/src/store_write.rs

```diff
@@ -453,6 +453,11 @@ impl IndexedBucket {
                 *is_sorted = false;
             }
         }
+        if let Some(row_id) = col_row_id.last() {
+            if row.row_id() < *row_id {
+                *is_sorted = false;
```

Comment:
This is _slightly_ over-pessimistic. If the last row has an older time, then it doesn't matter what the row-id is. Replacing both `if` statements with this should be correct:

```rs
        if let (Some(last_time), Some(last_row_id)) = (col_time.last(), col_row_id.last()) {
            // NOTE: Within a single timestamp, we use the Row ID as tie-breaker
            *is_sorted &= (last_time, last_row_id) <= (time.as_i64(), row.row_id());
        }
```